### PR TITLE
Revoke tokens upon user deactivation

### LINKED ISSUE #776: [user deactivation] revoke all tokens
Part of the [User Deactivation Epic](https://github.com/fabric8-services/fabric8-auth/issues/774)

All tokens of the user to deactivate should be revoked.

### PR DESCRIPTION
**Pull request by xcoulon**
Fixes #776

### CODE REVIEW

**Comment by xcoulon** [comment]
Based on PR for #775. Will rebase against `master` once this latter PR is merged.

File: Gopkg.lock [hunk 1]
```diff
@@ -2,139 +2,104 @@
 
 
 [[projects]]
-  digest = "1:2aff5edb9bccd2974090fddb17ca7ab05a3f5c983db567c30c7f0b53404f5783"
```
Comment (alexeykazakov) [inline Gopkg.lock]: What dep version are you using? Looks like something <0.5?

File: authentication/account/service/user_blackbox_test.go [hunk 2]
```diff
@@ -66,28 +66,51 @@ func (s *userServiceBlackboxTestSuite) TestDeprovision() {
...
+			assert.Equal(t, tok.Token().Status, token.TOKEN_STATUS_VALID)
+		}
+
 	})
 
+	s.T().Run("not found", func(t *testing.T) {
+		// when
+		_, err := s.Application.UserService().DeactivateUser(s.Ctx, "unknown")
+		// then
+		require.Error(t, err)
```
Comment (alexeykazakov) [inline authentication/account/service/user_blackbox_test.go]: Check the error type and message too.

File: authentication/account/service/user_blackbox_test.go [hunk 3]
```diff
@@ -66,28 +66,51 @@ func (s *userServiceBlackboxTestSuite) TestDeprovision() {
...
+			assert.Equal(t, tok.Token().Status, token.TOKEN_STATUS_REVOKED)
+		}
+		// lastly, verify that user to keep intact remainded intact
 		loadedUser = s.Graph.LoadUser(userToStayIntact.IdentityID())
 		assert.True(t, loadedUser.User().Active)
 		testsupport.AssertIdentityEqual(t, userToStayIntact.Identity(), loadedUser.Identity())
+		for _, tID := range []uuid.UUID{token3.TokenID(), token4.TokenID()} {
+			tok := s.Graph.LoadToken(tID)
+			require.NotNil(t, tok)
+			assert.Equal(t, tok.Token().Status, token.TOKEN_STATUS_VALID)
```
Comment (sbryzak) [inline authentication/account/service/user_blackbox_test.go]: You can do this easier:

`assert.True(t, tok.Token().Valid())`

File: authorization/token/token.go [hunk 4]
```diff
@@ -28,6 +29,7 @@ const (
 
 	// Token Statuses
 
+	TOKEN_STATUS_VALID         = 0
```
Comment (sbryzak) [inline authorization/token/token.go]: I'd prefer not to define such a constant - a valid token is a token that has none of the other status flags, we have a function which abstracts this test in the `Token` repository object - `Valid()`, which is a preferable way to test token validity.

File: Gopkg.lock (hunk 1 above)
Comment (xcoulon) [reply Gopkg.lock]: no, I'm using `0.5.0`:
```
$ dep version
dep:
 version     : v0.5.0
 build date  : 2018-07-26
 git hash    : 224a564
 go version  : go1.10.3
 go compiler : gc
 platform    : darwin/amd64
 features    : ImportDuringSolve=false
```

File: authorization/token/token.go (hunk 4 above)
Comment (xcoulon) [reply authorization/token/token.go]: ok now that I know I can use `tok.Token().Valid()` as you showed above. Although, the status flag being `0` by default, I don't see a problem with having a constant to reflect this value. Anyways, I'll remove it for now.

File: authentication/account/service/user_blackbox_test.go (hunk 3 above)
Comment (xcoulon) [reply authentication/account/service/user_blackbox_test.go]: ah cool, thanks @sbryzak !
change applied in be07719

File: authentication/account/service/user_blackbox_test.go (hunk 2 above)
Comment (xcoulon) [reply authentication/account/service/user_blackbox_test.go]: done in 73cb2d5

**Comment by xcoulon** [comment]
thanks for the approval, @sbryzak 👍

**Comment by xcoulon** [comment]
thanks for the approval too, @alexeykazakov ;)

**Comment by xcoulon** [comment]
[test]

**Pull request merged**